python/sglang/srt/reasoning_parser.py

```diff
@@ -0,0 +1,93 @@
+import json
+import logging
+import re
+from typing import Any, Dict, List, Optional, Tuple, Type
+
+
+class BaseReasoningParser:
+    """Base class for reasoning parser."""
+
+    def __init__(self, think_start_token: str, think_end_token: str):
+        self._buffer = ""
+        self.think_start_token = think_start_token
+        self.think_end_token = think_end_token
+        self.pattern = re.compile(
+            rf"{self.think_start_token}(.*?){self.think_end_token}", re.DOTALL
```

Comment:
The most recent tokenizer hardcodes the opening `<think>` tag: https://huggingface.co/deepseek-ai/DeepSeek-R1/commit/8a58a132790c9935686eb97f042afa8013451c9f

This means the text coming back from inference won't include `<think>`, this is why I updated #3202 to assume the model is reasoning until `</think>` is seen, it also strips out `<think>` to handle the old chat template.

Reply:
@tot0 

The PR added the start token if it is missing:

```
            # Add the start token to the beginning of the text.
            text = self.think_start_token + text
```

You can see it in `detect_and_parse`